### PR TITLE
Brms 311

### PR DESCRIPTION
**Pull request by manishsw**

### CODE REVIEW

File: app/uk/gov/hmrc/brm/config/BrmConfig.scala [hunk 1]
```diff
@@ -48,18 +48,9 @@ trait BrmConfig extends ServicesConfig {
   def ignoreMiddleNames : Boolean = getConfBool("birth-registration-matching.matching.ignoreMiddleNames",
```
Comment (adamconder) [inline app/uk/gov/hmrc/brm/config/BrmConfig.scala]: @manishsw Can we please rename this switch to ignoreAdditionalNames as per the interface.

File: app/uk/gov/hmrc/brm/connectors/NRSConnector.scala [hunk 2]
```diff
@@ -70,21 +68,11 @@ class NRSConnector(var httpPost: HttpPost = WSHttp, auditor: ScotlandAudit = new
       (detailsUri, Json.parse(
         s"""
            |{
-           | "$JSON_FIRSTNAME_PATH" : "${firstName(fName,aName)}",
+           | "$JSON_FIRSTNAME_PATH" : "${CommonUtil.forname(fName,aName)}",
```
Comment (adamconder) [inline app/uk/gov/hmrc/brm/connectors/NRSConnector.scala]: @manishsw should be forenames

File: app/uk/gov/hmrc/brm/connectors/NRSConnector.scala [hunk 3]
```diff
@@ -70,21 +68,11 @@ class NRSConnector(var httpPost: HttpPost = WSHttp, auditor: ScotlandAudit = new
...
            |}
          """.stripMargin))
   }
 
-
-  //add additional name to firstname while sending to respective service.
-  protected def firstName(firstName: String, additionalName: Option[String]): String = {
-    val forenames = BrmConfig.ignoreAdditionalName match {
-      case true => NameFormat(firstName)
-      case false => NameFormat(firstName).concat(" ").concat(NameFormat(additionalName.getOrElse(""))).trim
```
Comment (adamconder) [inline app/uk/gov/hmrc/brm/connectors/NRSConnector.scala]: @manishsw could be s"${NameFormat(firstName)} ${NameFormat(additionalName.getOrElse(""))}".trim

File: app/uk/gov/hmrc/brm/services/MatchingAlgorithm.scala [hunk 4]
```diff
@@ -66,11 +66,8 @@ trait MatchingAlgorithm {
   }
 
   protected[MatchingAlgorithm] def matchFirstNames(payload: Payload, record: Record) : Match = {
-    //add additonal name to firstname based on feature toggle.
-   val firstNamePayload =  ignoreAdditionalName match {
-      case true =>  payload.firstName.names.listToString
-      case false =>  concatAdditionalName(payload).names.listToString
-    }
+    //add additonal name to firstname based on feature toggle value
+    val firstNamePayload =  forname(payload.firstName, payload.additionalNames).names.listToString
```
Comment (adamconder) [inline app/uk/gov/hmrc/brm/services/MatchingAlgorithm.scala]: @manishsw forenames

File: app/uk/gov/hmrc/brm/config/BrmConfig.scala (hunk 1 above)
Comment (manishsw) [reply app/uk/gov/hmrc/brm/config/BrmConfig.scala]: Done.

File: app/uk/gov/hmrc/brm/connectors/NRSConnector.scala (hunk 3 above)
Comment (manishsw) [reply app/uk/gov/hmrc/brm/connectors/NRSConnector.scala]: Done

**Comment by manishsw** [comment]
@adamconder  Done changes , kindly merge if ok.

**Pull request merged**